src/features/identityCheck/api/useEduConnectLogin.ts

```diff
@@ -21,27 +20,22 @@ export function useEduConnectLogin() {
       if (response.ok) {
         const finalURL = response.headers.get('educonnect-redirect')
         if (finalURL) {
-          setLoginUrl(finalURL)
-          return
+          return finalURL
         }
       }
       setError(new Error('Failed to get EduConnect login url'))
+      return
     } catch (err) {
       setError(err as Error | null)
+      return
     }
   }, [])
```

Comment:
Pas besoin des 2 `return` si la fonction est finie juste après, non ?

Reply:
Si je ne les mets pas typescript râle comme ceci "Not all code paths return a value." 😬

Reply:
En effet !
Dans ce cas, je pense qu'on peut simplifier en en mettant qu'un seul, après le catch (ligne 31.5), il me semble

Reply:
Effectivement! Je corrige